platform/web/js/libs/library_godot_os.js

```diff
@@ -346,6 +346,17 @@ const GodotOS = {
 		return concurrency < 2 ? concurrency : 2;
 	},
 
+	godot_js_os_pthread_pool_size_get__proxy: 'sync',
+	godot_js_os_pthread_pool_size_get__sig: 'i',
+	godot_js_os_pthread_pool_size_get: function () {
+		if (typeof PThread === 'undefined') {
+			// Threads aren't supported, so default to `1`.
+			return 1;
+		}
+		const total = PThread.runningWorkers.length + PThread.unusedWorkers.length;
+		return Math.max(1, total - 1);
```

Comment:
cc @Faless. Can you confirm that this avoids the problem you identified in https://github.com/godotengine/godot/pull/54499 It seems like we are querying the number of web workers allocated by emscriptem rather than just querying the number of processors. That sounds like the right solution to me, but I am not well versed in Javascript/emscriptem.

Bottom line, we can't use only one thread in the WTP. We generally need at least 3 (in the case the separate rendering thread and separate physics thread are enabled.

Reply:
I comment about this [here](https://github.com/godotengine/godot/pull/104458#discussion_r2008222268), but I'm still unsure since the actual thread pool is hardcoded to 8 even if the user has less processors.

What we can do though is change `-sPTHREAD_POOL_SIZE=8` to

```
-sPTHREAD_POOL_SIZE=Math.max(3, Math.min(8, navigator.hardwareConcurrency))
```

See: https://github.com/emscripten-core/emscripten/blob/36423eb7efbecfb901300dd65d3f06967fafd984/src/settings.js#L1667-L1675

Reply:
@clayjohn pushed with

```
-sPTHREAD_POOL_SIZE=Math.min(8,navigator.hardwareConcurrency)
```
Which results in the following emscripten code:
```js
var pthreadPoolSize=Math.min(8,navigator.hardwareConcurrency)
```

I removed the `Math.max(3,...)`  because if the user does not have 3 cores, then it doesn't matter whether it's 3 or 1 the threadpool size. If the minimum required for multithreading is 3, the engine should fallback to single threaded, but I think that's outside of the scope of this PR (not sure whether that logic is implemented already)